### PR TITLE
env hack to deal with sccache issues

### PR DESCRIPTION
**Pull request by timotheyca**
What this does: adds `CARGOZIGBUILD_RUSTC_VERSION` (idk what else to name it) variable that is used by `Zig::execute_compiler` to avoid calling `sccache rustc -vV` in some cases

### CODE REVIEW

**Comment by timotheyca** [comment]
addresses https://github.com/rust-cross/cargo-zigbuild/issues/295

File: src/zig.rs [hunk 1]
```diff
@@ -106,7 +106,10 @@ impl Zig {
             .and_then(|index| cmd_args.get(index + 1));
         let target_info = TargetInfo::new(target);
 
-        let rustc_ver = rustc_version::version()?;
+        let rustc_ver = match env::var("CARGOZIGBUILD_RUSTC_VERSION") {
```
Comment (messense) [inline src/zig.rs:109]:
```suggestion
        let rustc_ver = match env::var("CARGO_ZIGBUILD_RUSTC_VERSION") {
```

File: src/zig.rs [hunk 2]
```diff
@@ -512,6 +515,11 @@ impl Zig {
             .map(|target| target.split_once('.').map(|(t, _)| t).unwrap_or(target))
             .collect::<Vec<&str>>();
         let rustc_meta = rustc_version::version_meta()?;
+        Self::add_env_if_missing(
+            cmd,
+            "CARGOZIGBUILD_RUSTC_VERSION",
```
Comment (messense) [inline src/zig.rs:520]:
```suggestion
            "CARGO_ZIGBUILD_RUSTC_VERSION",
```

**Review by messense — COMMENTED** [review]
Thanks, let's call this `CARGO_ZIGBUILD_RUSTC_VERSION`.

**Review by messense — APPROVED** [review]
I'll do the rename locally

**Pull request merged**